### PR TITLE
VACMS-11334: Implement VBA permissions and Restricted archive workflow

### PR DESCRIPTION
**Pull request by omahane**
## Description

Relates to #11334

## Testing done
Manually

## Screenshots
Screenshot 2023-10-27 at 10 13 01 AM (2)

## QA steps
_Note: When steps are first introduced in the following QA, they are more explicit. After the first mention of an action, subsequent actions are bundled. The steps around saving a node, for instance, follow this pattern._

#### Set up the local VBA facility editor
_Note: This should already be done, unless the environment has refreshed._

As an admin, do the following 
- [x] Create the **local.vba.facility.editor@va.gov** (username and email) user
- [x] Assign user the "Content Publisher" and "Content creator - VBA" role 
- [x] Assign them to the "Cheyenne VA Regional Benefit office" **Section**
- [x] Create a "Pension" VBA Facility Service 
  - [x] In the "Cheyenne VA Regional Benefit office" **Section**
  - [x] At the "Cheyenne VA Regional Benefit office" **Facility**
- [x] Save the "Pension" VBA Facility Service as a draft

#### Test Permissions of the local VBA facility editor
As local.vba.facility.editor@va.gov, do the following
- [ ] Add content
  - [ ] Confirm that "VBA Facility Service" is the only option for adding content
  - [ ] Choose "VBA Facility Service"
  - [ ] Confirm that the **Section** is already set to "Cheyenne VA Regional Benefit Office" 
  - [ ] Select "Cheyenne VA Regional Benefit Office" in **Facility**
  - [ ] Select "Home loans" in **Service name**
  - [ ] **Save as** Published
  - [ ] Add a **Revision log message**
  - [ ] Click **Save**
  - [ ] Confirm that the service was saved
- [ ] Edit a VBA Facility Service someone else made
  - [ ] Go to the "Cheyenne VA Regional Benefit Office - Pension" service
  - [ ] Edit the node
    - [ ] Change the service location by adding an office name to **Name of office or location**
    - [ ] Publish the service
    - [ ] Confirm that you can publish the service
    - [ ] Confirm that the office name you added is visible
- [ ] Edit a VBA Facility
  - [ ] Edit the "[Cheyenne VA Regional Benefit Office](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/node/4338/edit)"
  - [ ] Set the **Operating status** to "Facility notice"
  - [ ] Add "Test message" to **Operating status - more info**
  - [ ] Save the changes as "Draft"
  - [ ] Confirm that the facility's name has changed
  - [ ] Edit the facility, saving as "Published"

#### Set up the national VBA facility editor
_Note: This should already be done, unless the environment has refreshed._

As an admin, do the following 
- [x] Create the **national.vba.facility.editor@va.gov** (username and email) user
- [x] Assign user the "Content Publisher" and "Content creator - VBA" role 
- [x] Assign them to the "VBA Facilities" and "VBA Service Regions" **Sections**

#### Test permissions of national VBA facility editor
As national.vba.facility.editor@va.gov, do the following
- [x] Create content in two different sections
  - [x] Add content in "Cheyenne VA Regional Benefit Office" Section
    - [x] Confirm that "VBA Facility Service" is the only option for adding content
    - [x] Choose "VBA Facility Service"
    - [x] Select "Cheyenne VA Regional Benefit Office" In the **Section**
    - [x] Select "Cheyenne VA Regional Benefit Office" in **Facility**
    - [x] Select "Veterans Pension" in **Service name**
    - [x] Publish the node
    - [x] Confirm that the node published
  - [x] Add content in "Baltimore VA Regional Benefit Office" Section
    - [x] Confirm that "VBA Facility Service" is the only option for adding content
    - [x] Choose "VBA Facility Service"
    - [x] Select "Baltimore VA Regional Benefit Office" In the **Section**
    - [x] Select "Baltimore VA Regional Benefit Office" in **Facility**
    - [x] Select "Pension" in **Service name**
    - [x] Publish the node
    - [ ] Confirm that the node published
- [x] Edit content in two different sections
  - [ ] Edit the "[Baltimore VA Regional Benefit Office - Pension](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/node/62689/edit)"
    - [x] Change the service location by adding an office name to **Name of office or location**
    - [x] Publish the service
    - [x] Confirm that the node was published with the change
  - [x] Edit the "[Pre-Discharge Site at Offutt Air Force Base](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/node/4159/edit)"
    - [x] Add test copy to the **Page introduction**
    - [x] Publish the facility
    - [x] Confirm that the node was published with the added "Page introduction"
- [x] Edit a Service region
  - [x] Edit the [Regional Loan Center in Cleveland](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/node/61786/edit) 
    - [x] Add an email address under **Contact Info**/**Email**
      - [x] Add "mytestemail@va.gov" to **Email address**
      - [x] Add "Service region representative" to **Label**
      - [x] Click **Create email address**
      - [x] Save as "Published"
      - [x] Confirm that the node was published with the added email address

# Test views not covered by the patch:
As an admin, check the moderation state filter on these Views:
  - [x]  [Promos](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/admin/content/promos) (uses blocks)
      - [x] Appears unaffected 
  - [x]  [Moderated content](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/admin/content/moderated) (uses revisions)
    - [x] Appears to only have 2 moderation states
  - [x]  [VAMC top task page revision histories](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/admin/content/audit/vamc-top-task-page-revision-histories) (uses revisions)
    - [x] Appears to have moderation workflows separated 
  - [x]  [Content search] https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/admin/content/audit/search (uses grouped filter across content types)
    - [x] Appears unaffected 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### CODE REVIEW

**Comment by davidmpickett** [comment]
Not sure if this is new, or a different flavor of #12737, but editors shouldn't be able to create Services for Facilities they don't permission to edit otherwise.

Screenshot 2023-10-30 081002

screencapture-cms-uuviupwhkcoop9okwkjb5fw3j1rdyqtj-ci-cms-va-gov-node-62767-2023-10-30-08_15_00

**Comment by davidmpickett** [comment]
Actually looks like @omahane [noted the lack of a winnower](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11197#issuecomment-1289648669) as a limitation back in #11197. At that point there weren't enough Sections for VBA to set up winnowing logic beyond just show only VBA facilities. 

Updating the winnowing is probably out of scope for #11334, so probably needs a new ticket

File: tests/phpunit/Security/RolesPermissionsTest.php [hunk 1]
```diff
@@ -252,6 +259,12 @@ public function expectedPerms() {
           'view node link report',
         ],
       ],
+      [
+        'content_creator_vba',
+        [
+          'create vba_facility_service content',
```
Comment (swirtSJW) [inline tests/phpunit/Security/RolesPermissionsTest.php:265]: Shouldn't this also have 'view node link repor'?

**Review by swirtSJW — COMMENTED** [review]
Scanning of the code changes look good.  Just one question to come out of that before QA.

File: config/sync/core.entity_view_display.node.landing_page.support_services_listing.yml [hunk 2]
```diff
@@ -77,6 +77,18 @@ targetEntityType: node
 bundle: landing_page
 mode: support_services_listing
 content:
+  breadcrumbs:
```
[the remark this answers is not in the document]
Comment (omahane) [reply config/sync/core.entity_view_display.node.landing_page.support_services_listing.yml]: I've removed this change.

File: tests/phpunit/Security/RolesPermissionsTest.php (hunk 1 above)
Comment (omahane) [reply tests/phpunit/Security/RolesPermissionsTest.php:265]: I've added this permission

**Comment by swirtSJW** [comment]
Flow looks good
https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/admin/reports/system/workflow/content_moderation/restricted_archive/all
image

**Review by swirtSJW — APPROVED** [review]
Testing steps all checkout.  Nice work @omahane on the PR and the testing.

The 4 cases that are not using the multiworkflow filter seem fine.  They are not broken and their "clunkiness" is understandable given the scope of the limited purpose of those Views.

Check with @jilladams and @xiongjaneg about wither this PR needs a review from CMS platform.   I am thinking it likely does but I defer to them.

**Comment by davidmpickett** [comment]
Copying QA steps for my own runthrough

### Test local VBA facility editor:

#### Set up the local VBA facility editor
_Note: This should already be done, unless the environment has refreshed._

As an admin, do the following 
- [x] Create the **local.vba.facility.editor@va.gov** (username and email) user
- [x] Assign user the "Content Publisher" and "Content creator - VBA" role 
- [x] Assign them to the "Cheyenne VA Regional Benefit office" **Section**
- [x] Create a "Pension" VBA Facility Service 
  - [x] In the "Cheyenne VA Regional Benefit office" **Section**
  - [x] At the "Cheyenne VA Regional Benefit office" **Facility**
- [x] Save the "Pension" VBA Facility Service as a draft

#### Test Permissions of the local VBA facility editor
As local.vba.facility.editor@va.gov, do the following
- [ ] Add content
  - [ ] Confirm that "VBA Facility Service" is the only option for adding content
  - [ ] Choose "VBA Facility Service"
  - [ ] Confirm that the **Section** is already set to "Cheyenne VA Regional Benefit Office" 
  - [ ] Select "Cheyenne VA Regional Benefit Office" in **Facility**
  - [ ] Select "Home loans" in **Service name**
  - [ ] **Save as** Published
  - [ ] Add a **Revision log message**
  - [ ] Click **Save**
  - [ ] Confirm that the service was saved
- [ ] Edit a VBA Facility Service someone else made
  - [ ] Go to the "Cheyenne VA Regional Benefit Office - Pension" service
  - [ ] Edit the node
    - [ ] Change the service location by adding an office name to **Name of office or location**
    - [ ] Publish the service
    - [ ] Confirm that you can publish the service
    - [ ] Confirm that the office name you added is visible
- [ ] Edit a VBA Facility
  - [ ] Edit the "[Cheyenne VA Regional Benefit Office](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/node/4338/edit)"
  - [ ] Set the **Operating status** to "Facility notice"
  - [ ] Add "Test message" to **Operating status - more info**
  - [ ] Save the changes as "Draft"
  - [ ] Confirm that the facility's name has changed
  - [ ] Edit the facility, saving as "Published"

### Test National VBA facility editor

#### Set up the national VBA facility editor
_Note: This should already be done, unless the environment has refreshed._

As an admin, do the following 
- [x] Create the **national.vba.facility.editor@va.gov** (username and email) user
- [x] Assign user the "Content Publisher" and "Content creator - VBA" role 
- [x] Assign them to the "VBA Facilities" and "VBA Service Regions" **Sections**

#### Test permissions of national VBA facility editor
As national.vba.facility.editor@va.gov, do the following
- [x] Create content in two different sections
  - [x] Add content in "Cheyenne VA Regional Benefit Office" Section
    - [x] Confirm that "VBA Facility Service" is the only option for adding content
    - [x] Choose "VBA Facility Service"
    - [x] Select "Cheyenne VA Regional Benefit Office" In the **Section**
    - [x] Select "Cheyenne VA Regional Benefit Office" in **Facility**
    - [x] Select "Veterans Pension" in **Service name**
    - [x] Publish the node
    - [x] Confirm that the node published
  - [x] Add content in "Baltimore VA Regional Benefit Office" Section
    - [x] Confirm that "VBA Facility Service" is the only option for adding content
    - [x] Choose "VBA Facility Service"
    - [x] Select "Baltimore VA Regional Benefit Office" In the **Section**
    - [x] Select "Baltimore VA Regional Benefit Office" in **Facility**
    - [x] Select "Pension" in **Service name**
    - [x] Publish the node
    - [ ] Confirm that the node published
- [x] Edit content in two different sections
  - [ ] Edit the "[Baltimore VA Regional Benefit Office - Pension](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/node/62689/edit)"
    - [x] Change the service location by adding an office name to **Name of office or location**
    - [x] Publish the service
    - [x] Confirm that the node was published with the change
  - [x] Edit the "[Pre-Discharge Site at Offutt Air Force Base](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/node/4159/edit)"
    - [x] Add test copy to the **Page introduction**
    - [x] Publish the facility
    - [x] Confirm that the node was published with the added "Page introduction"
- [x] Edit a Service region
  - [x] Edit the [Regional Loan Center in Cleveland](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/node/61786/edit) 
    - [x] Add an email address under **Contact Info**/**Email**
      - [x] Add "mytestemail@va.gov" to **Email address**
      - [x] Add "Service region representative" to **Label**
      - [x] Click **Create email address**
      - [x] Save as "Published"
      - [x] Confirm that the node was published with the added email address

# Test views not covered by the patch:
As an admin, check the moderation state filter on these Views:
  - [x]  [Promos](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/admin/content/promos) (uses blocks)
      - [x] Appears unaffected 
  - [x]  [Moderated content](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/admin/content/moderated) (uses revisions)
    - [x] Appears to only have 2 moderation states
  - [x]  [VAMC top task page revision histories](https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/admin/content/audit/vamc-top-task-page-revision-histories) (uses revisions)
    - [x] Appears to have moderation workflows separated 
  - [x]  [Content search] https://pr15872-uuviupwhkcoop9okwkjb5fw3j1rdyqtj.ci.cms.va.gov/admin/content/audit/search (uses grouped filter across content types)
    - [x] Appears unaffected

**Comment by davidmpickett** [comment]
Thumbs up on QA from me

**Review by davidmpickett — APPROVED** [review]
From a functionality perspective, this is working as intended and QA was successfull

**Pull request merged**